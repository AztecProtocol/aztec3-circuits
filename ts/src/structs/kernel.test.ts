import { expectSerializeToMatchSnapshot } from "../tests/expectSerializeToMatchSnapshot.js";
import { makePreviousKernelData } from "../tests/factories.js";

describe("structs/kernel", () => {
  it(`serializes and prints previous_kernel_data`, async () => {
    const previousKernelData = makePreviousKernelData();
    await expectSerializeToMatchSnapshot(
      previousKernelData.toBuffer(),
      "abis__test_roundtrip_serialize_previous_kernel_data"
    );
  });
});
