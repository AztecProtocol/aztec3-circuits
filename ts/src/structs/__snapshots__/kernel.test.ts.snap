// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`structs/kernel serializes and prints previous_kernel_data 1`] = `
"public_inputs: end:
aggregation_object:
P0: { 0x100000002000000000000000000000000000000000000000000000000, 0x3000000000000000000000000 }
P1: { 0x4000000000000000000000000, 0x5000000000000000000000000 }
public_inputs:
[
]
proof_witness_indices:
[ ]
has_data: 0

private_call_count: 0x6000000060000000000000000000000000000000000
new_commitments:
[ 0xb0000000c00000000000000000000000000000000000000000000000000 0xd000000000000000000000000000000000000000000000000 0x101000000000000000000000000000000000000000000000000 0x102000000000000000000000000000000000000000000000000 0x103000000000000000000000000000000000000000000000000 0x104000000000000000000000000000000000000000000000000 0x105000000000000000000000000000000000000000000000000 0x106000000000000000000000000000000000000000000000000 0x107000000000000000000000000000000000000000000000000 0x108000000000000000000000000000000000000000000000000 0x109000000000000000000000000000000000000000000000000 0x10a000000000000000000000000000000000000000000000000 0x10b000000000000000000000000000000000000000000000000 0x10c000000000000000000000000000000000000000000000000 0x10d000000000000000000000000000000000000000000000000 0x10e000000000000000000000000000000000000000000000000 ]
new_nullifiers:
[ 0x10f000000000000000000000000000000000000000000000000 0x110000000000000000000000000000000000000000000000000 0x201000000000000000000000000000000000000000000000000 0x202000000000000000000000000000000000000000000000000 0x203000000000000000000000000000000000000000000000000 0x204000000000000000000000000000000000000000000000000 0x205000000000000000000000000000000000000000000000000 0x206000000000000000000000000000000000000000000000000 0x207000000000000000000000000000000000000000000000000 0x208000000000000000000000000000000000000000000000000 0x209000000000000000000000000000000000000000000000000 0x20a000000000000000000000000000000000000000000000000 0x20b000000000000000000000000000000000000000000000000 0x20c000000000000000000000000000000000000000000000000 0x20d000000000000000000000000000000000000000000000000 0x20e000000000000000000000000000000000000000000000000 ]
private_call_stack:
[ 0x20f000000000000000000000000000000000000000000000000 0x210000000000000000000000000000000000000000000000000 0x301000000000000000000000000000000000000000000000000 0x302000000000000000000000000000000000000000000000000 0x303000000000000000000000000000000000000000000000000 0x304000000000000000000000000000000000000000000000000 0x305000000000000000000000000000000000000000000000000 0x306000000000000000000000000000000000000000000000000 ]
public_call_stack:
[ 0x307000000000000000000000000000000000000000000000000 0x308000000000000000000000000000000000000000000000000 0x401000000000000000000000000000000000000000000000000 0x402000000000000000000000000000000000000000000000000 0x403000000000000000000000000000000000000000000000000 0x404000000000000000000000000000000000000000000000000 0x405000000000000000000000000000000000000000000000000 0x406000000000000000000000000000000000000000000000000 ]
l1_msg_stack:
[ 0x407000000000000000000000000000000000000000000000000 0x408000000000000000000000000000000000000000000000000 0x501000000000000000000000000000000000000000000000000 0x502000000000000000000000000000000000000000000000000 ]
new_contracts:
[ contract_address: 0x503000000000000000000000000000000000000000000000000
portal_contract_address: 0x504000000000000000000000000000000000000000000000000
function_tree_root: 0x601000000000000000000000000000000000000000000000000
 contract_address: 0x202020202020200000000000000000000000000000000000000000000000000
portal_contract_address: 0x603000000000000000000000000000000000000000000000000
function_tree_root: 0x602000000000000000000000000000000000000000000000000
 contract_address: 0x303030303030300000000000000000000000000000000000000000000000000
portal_contract_address: 0x604000000000000000000000000000000000000000000000000
function_tree_root: 0x603000000000000000000000000000000000000000000000000
 contract_address: 0x404040404040400000000000000000000000000000000000000000000000000
portal_contract_address: 0x605000000000000000000000000000000000000000000000000
function_tree_root: 0x604000000000000000000000000000000000000000000000000
 contract_address: 0x505050505050500000000000000000000000000000000000000000000000000
portal_contract_address: 0x606000000000000000000000000000000000000000000000000
function_tree_root: 0x605000000000000000000000000000000000000000000000000
 contract_address: 0x606060606060600000000000000000000000000000000000000000000000000
portal_contract_address: 0x607000000000000000000000000000000000000000000000000
function_tree_root: 0x606000000000000000000000000000000000000000000000000
 contract_address: 0x707070707070700000000000000000000000000000000000000000000000000
portal_contract_address: 0x608000000000000000000000000000000000000000000000000
function_tree_root: 0x607000000000000000000000000000000000000000000000000
 contract_address: 0x808080808080800000000000000000000000000000000000000000000000000
portal_contract_address: 0x609000000000000000000000000000000000000000000000000
function_tree_root: 0x608000000000000000000000000000000000000000000000000
 ]
optionally_revealed_data:
[ call_stack_item_hash: 0x909090909090900000000000000000000000000000000000000000000000000
function_data:
function_selector: 0
is_private: 0
is_constructor: 0

emitted_events:
[ 0x60a000000000000000000000000000000000000000000000000000000000000 0x701000007020100000000000000000000000000000000000000000000000000 0x801000000000000000000000000000000000000000000000000 0x802000000000000000000000000000000000000000000000000 ]
vk_hash: 0x803000000000000000000000000000000000000000000000000
portal_contract_address: 0x804000000000000000000000000000000000000000000000000
pay_fee_from_l1: 0
pay_fee_from_public_l2: 0
called_from_l1: 0
called_from_public_l2: 0
 call_stack_item_hash: 0x70300000000000000000000000000000000000000000000000000000000
function_data:
function_selector: 4040404
is_private: 1
is_constructor: 0

emitted_events:
[ 0x100000000000000000000000000000000000000000000000000000000000000 0x702000007030100000000000000000000000000000000000000000000000000 0x802000000000000000000000000000000000000000000000000 0x803000000000000000000000000000000000000000000000000 ]
vk_hash: 0x804000000000000000000000000000000000000000000000000
portal_contract_address: 0x805000000000000000000000000000000000000000000000000
pay_fee_from_l1: 0
pay_fee_from_public_l2: 0
called_from_l1: 0
called_from_public_l2: 0
 call_stack_item_hash: 0x70400000000000000000000000000000000000000000000000000000000
function_data:
function_selector: 5050505
is_private: 1
is_constructor: 0

emitted_events:
[ 0x100000000000000000000000000000000000000000000000000000000000000 0x703000007040100000000000000000000000000000000000000000000000000 0x803000000000000000000000000000000000000000000000000 0x804000000000000000000000000000000000000000000000000 ]
vk_hash: 0x805000000000000000000000000000000000000000000000000
portal_contract_address: 0x806000000000000000000000000000000000000000000000000
pay_fee_from_l1: 0
pay_fee_from_public_l2: 0
called_from_l1: 0
called_from_public_l2: 0
 call_stack_item_hash: 0x70500000000000000000000000000000000000000000000000000000000
function_data:
function_selector: 6060606
is_private: 1
is_constructor: 0

emitted_events:
[ 0x100000000000000000000000000000000000000000000000000000000000000 0x704000007050100000000000000000000000000000000000000000000000000 0x804000000000000000000000000000000000000000000000000 0x805000000000000000000000000000000000000000000000000 ]
vk_hash: 0x806000000000000000000000000000000000000000000000000
portal_contract_address: 0x807000000000000000000000000000000000000000000000000
pay_fee_from_l1: 0
pay_fee_from_public_l2: 0
called_from_l1: 0
called_from_public_l2: 0
 ]

constants:
old_tree_roots: private_data_tree_root: 0x70600000000000000000000000000000000000000000000000000000000
nullifier_tree_root: 0x707070701000100000000000000000000000000000000000000000000000000
contract_tree_root: 0x101000000000000000000000000000000000000000000000000
private_kernel_vk_tree_root: 0x102000000000000000000000000000000000000000000000000

tx_context: is_fee_payment_tx: 0
is_rebate_payment_tx: 0
is_contract_deployment_tx: 0
contract_deployment_data: 
constructor_vk_hash: 0x103000000000000000000000000000000000000000000000000000000
function_tree_root: 0x104000001000000000000000000000000000000000000000000000000
contract_address_salt: 0x105000000000000000000000000000000000000000000000000
portal_contract_address: 0x106000000000000000000000000000000000000000000000000


is_private: 0

proof: [ ]
vk: 0
vk_index: 67328
vk_path: [ 0x10801 0x1081818181818200000000000000000000000000000000000000000000 0x10000000000000000000 ]
"
`;
